content/blog/example-of-mapreduce-with-ezmeral-data-fabric-mapr-database-binary-table.md

```diff
@@ -0,0 +1,365 @@
+
+maprcli volume create -name test.binarytable1 -path /testbinarytable1volume
+
+# sudo -u mapr maprcli volume mount -name test.binarytable1 -path /testbinarytable1volume
+sudo -u mapr hadoop fs -ls -d -h /testbinarytable1volume
+sudo -u mapr hadoop mfs -ls /testbinarytable1volume
+```
+
+
+
+☝ The volume's name is: test.binarytable1 and it will be mounted as <ins>/testbinarytable1volume/</ins> in the EDF File System.
+
+
+
+#### Now we can create the Binary Table
+
+
+
+👇 Inside `hbase shell`:
+
+
+
+```
+create '/testbinarytable1volume/notifications','attributes','metrics'
+```
+
+
+
+Table name is <ins>/testbinarytable1volume/notifications</ins>. **attributes** and **metrics** are column families.
+
+
+
+**Note**: In EDF Database Binary Table, the table name is by default a path in the File System.
+You can change the style of the table name to be like in Apache HBase, refer to👉: [Mapping to HBase Table Namespaces][MappingToHbaseTableNamespaces].
+
+
+
+### Build and Run The MapReduce Application
+
+
+
+**Important note**: Before running the MapReduce application, if you are using the Development Environment, you have to complete the following steps: [Installing Hadoop and YARN][InstallingHadoopAndYarn].
+
+
+It's because by default the YARN framework is not installed and hence the MapReduce framework is not installed.
+
+
+
+#### MapReduce Application Source Code
+
+
+
+I have put the code on Github: [Example-MapReduce-With-EzmeralDataFabricMapR-DatabaseBinaryTable][MapReduceEDFDataBaseBinaryTable].
+You can download it and compile it using Visual Studio Code.
+
+
+
+This MapReduce application is simple, the following is the logic:
+
+
+
+1. Get data from the source table: <ins>/testbinarytable1volume/notifications</ins> and output aggregated data to the target table: <ins>/testbinarytable1volume/summary</ins>.
+2. It's basically a varietas of "Word Count". This MapReduce application simply aggregates the number of rows which contains a column called "type" in the "attributes" column family.
+
+
+
+    For example, there may be comment type, promotion type, friend-request type, etc. in this table.
+    Then the app will count how many rows of comment type, promotion typee, friend-request type are there.
+
+
+
+#### How to Run it On EDF
+
+
+
+You can run the MapReduction application on your one-node cluster of EDF Development Environment, then.
+Refer to the following command:
+
+
+
+First, we need to create a new table(target table) for storing the counter number:
+
+
+
+```bash
+create '/testbinarytable1volume/summary','metrics'
+```
+
+
+
+Then, run the MapReduce application via `yarn jar` command:
+
+
+
+```bash
+sudo -u mapr \
+yarn jar ./target/original-hbase-example-1.0-SNAPSHOT.jar com.shouneng.learn.mapReduce.Main \
+  -libjar ./hbase-server-1.4.13.200-eep-810.jar,hbase-client-1.4.13.200-eep-810.jar,hadoop-common-2.7.6.200-eep-810.jar,hbase-common-1.4.13.203-eep-810.jar
+```
+
+
+
+Output:
+
+
+
+```

```

Comment:
Please, make sure to specify the language syntax "markdown" for the code snippet below.